rs/state_layout/src/state_layout.rs

```diff
@@ -2444,6 +2566,25 @@ fn copy_file_and_set_permissions(
         debug_assert!(false);
     }
 
+    copy_file_and_set_permissions(log, src, dst, dst_permissions)?;
+
+    // We keep the directory writable though to make sure we can rename
+    // them or delete the files.
```

Comment:
This comment seems out of context here.

Reply:
You're right. I tried to move the sentiment of the comment to somewhere more sensible.